src/app/loadout-drawer/subclass-drawer/Options.tsx

```diff
@@ -0,0 +1,183 @@
+                style={{ strokeDashoffset: 0 }}
+                fillOpacity="0"
+                stroke="#979797"
+                strokeWidth="1"
+                points="47,0 94,47 47,94 0,47"
+                strokeLinecap="butt"
+              />
+            </svg>
+          </div>
+        </div>
+      )}
+      <div className={styles.abilities}>
+        <Abilities abilities={abilities} selectedPlugs={selectedPlugs} dispatch={dispatch} />
+      </div>
+      <div className={styles.aspects}>
+        <Mods
+          mods={aspects}
+          selectedPlugs={selectedPlugs}
+          dispatch={dispatch}
+          onOpenPlugPicker={() => setShowPlugPicker(true)}
+        />
+      </div>
+      <div className={styles.fragments}>
+        <Mods
+          mods={fragments}
+          selectedPlugs={selectedPlugs}
+          maxSelectable={maxFragments}
+          dispatch={dispatch}
+          onOpenPlugPicker={() => setShowPlugPicker(true)}
+        />
+      </div>
+      {showPlugPicker &&
+        ReactDOM.createPortal(
+          <AspectAndFragmentDrawer
+            aspects={aspects}
+            fragments={fragments}
+            selectedPlugs={selectedPlugs}
+            dispatch={dispatch}
+            onClose={() => setShowPlugPicker(false)}
+          />,
+          document.body
+        )}
+    </div>
+  );
+}
+
+function getSocketsWithOptionsForCategory(
+  defs: D2ManifestDefinitions,
+  profileResponse: DestinyProfileResponse | undefined,
+  subclass: DimItem,
+  categoryHash: SocketCategoryHashes
+) {
+  const rtn: SocketWithOptions[] = [];
+  const title = defs.SocketCategory.get(categoryHash).displayProperties.name;
+  const indexes =
+    subclass.sockets?.categories.find((category) => category.category.hash === categoryHash)
+      ?.socketIndexes || [];
+  for (const index of indexes) {
+    const socket = subclass.sockets?.allSockets[index];
+    const plugSetHash = socket?.socketDefinition.reusablePlugSetHash;
+
+    if (plugSetHash && profileResponse) {
+      const plugSetItems = itemsForPlugSet(profileResponse, plugSetHash);
+
+      const options = _.uniqBy(
+        plugSetItems
+          ?.map((plugItem) => {
+            if (plugIsInsertable(plugItem)) {
+              return defs.InventoryItem.get(plugItem.plugItemHash);
+            }
+          })
+          .filter(isPluggableItem),
+        (option) => option.hash
+      );
+
+      if (socket && options?.length) {
+        const { plugCategoryHash } = options[0].plug;
+        rtn.push({ title, socket, options, plugCategoryHash });
+      }
+    }
+  }
+  return rtn;
+}
+
+function getSuperPlug(
+  defs: D2ManifestDefinitions,
+  subclass: DimItem,
+  categoryHash: SocketCategoryHashes
+) {
+  const indexes =
+    subclass.sockets?.categories.find((category) => category.category.hash === categoryHash)
+      ?.socketIndexes || [];
+  for (const index of indexes) {
+    const socket = subclass.sockets?.allSockets[index];
```

Comment:
This isn't really how socketIndexes works, as pointed out in the documentation for DimSockets: https://github.com/DestinyItemManager/DIM/blob/master/src/app/inventory/item-types.ts#L410-L414

Which is why there are a lot of helpers in https://github.com/DestinyItemManager/DIM/blob/master/src/app/utils/socket-utils.ts

Reply:
Ok, I was doing it off the Bungie API docs, I will have a look over those.

Reply:
Done, used a couple of helpers from here where I could.